coderd/workspaces.go

```diff
@@ -1188,6 +1191,7 @@ func convertWorkspace(
 		AutostartSchedule:                    autostartSchedule,
 		TTLMillis:                            ttlMillis,
 		LastUsedAt:                           workspace.LastUsedAt,
+		ImpendingDeletion:                    impendingDeletion,
```

Comment:
This field will show up on non-licensed workspace responses (although we won't do anything with it). 
Is this a concern? Should we rename to something more generic like `ImpendingCleanup`? I don't want to freak out anyone browsing our API. @sreya @bpmct

Reply:
So the value is populated, but on AGPL if that time comes, nothing happens right?

Reply:
Cleanup does sound better fwiw

Reply:
@Emyrk The value will be populated with an empty `time.Time` struct, so it will look like "0001-01-01T00:00:00Z".
I'm still concerned the name might make folks nervous. At the same time, cleanup is a bit ambiguous and could refer to a workspace stopping, which is also part of this RFC.

Reply:
Got it, if it is the 0 time, then I wouldn't worry about people misunderstanding it.

Reply:
That's not a bad idea. Is there any example of us doing that anywhere in the codebase?

Reply:
A couple of thoughts here:

- ImpendingCleanup feels a lot better
- It feels like `ImpendingCleanup` shouldn't have a value until 14 days before cleanup when the warning shows up in the dashboard/VS Code. To me, it feels more stressful seeing that every workspace has a cleanup date, although I also see the benefits in displaying this calculated value at all times. Curious if others have thoughts here..

Reply:
@bpmct 
Okay! I will change the name.
I was planning on dealing with the -14 day logic on the FE because IMO the  BE field should hold a controlled value, meaning we aren't going to change the way we calculate ImpendingDeletion, but we might very well change when we choose to alert the user. Also, alerting the user is more of a UI implementation detail. 
But I totally hear your point about alarming people. What do you think about Steven's suggestion of marshaling into a `nil` value?  Would that combined with a name change feel better?

Reply:
Chatting in Discord:

- will rename to `deletingAt`
- will make the field a `*time.Time` so it is`null` if unset
- not too concerned about AGPL case rn
- not too concerned about optics of having this field populated when `inactivityTTL` is set

Will make these changes in this PR